### DISCUSSION TITLE
Magento LTS latest compatibility?

### DISCUSSION THREAD

**Discussion by seansan** [OpenMage/magento-lts#2081] [Questions & Answers]
Magento LTS latest compatibility?

Would it be possible to list the "latest" (most updated) version of 
- PHP
- Mysql server
- Percona? MariaDB?
- Other

In the readme file that current Magento LTS version can run on without problems?

https://github.com/OpenMage/magento-lts/blob/1.9.3.x/README.md

I think this would be a great addition. And prevent any problems for new users ...

References
https://magento.stackexchange.com/questions/130828/is-ce-1-9-0-1-compatible-with-mysql-5-7
https://github.com/Inchoo/Inchoo_PHP7/wiki/PHP-7.1-and-7.2
https://community.magento.com/t5/Magento-1-x-Admin-Configuration/Magento-compatibility-with-MYSQL-5-7/td-p/56668

**Comment by Flyingmana** [comment]
I think thats only possible to say by having someone running it on this specific version.
Maybe we can maintain a List were people can contribute what versions they managed to run it on?

**Comment by seansan** [comment]
It would be great to make a start ... (I dont dare to upgrade mysql
currently ... )

I would be very interested in this information

**Comment by colinmollenhour** [comment]
Since Magento has maintained compatibility with such old versions of MySQL (5.0 if not older) and uses a DBAL that is very basic (read: limiting) in it's functionality, Magento is going to be compatible with basically any version of MySQL, Percona or MariaDb since these forks are newer than the oldest version supported. Between MySQL, Percona and MariaDb there are no significant differences in compatibility until after 5.6 where they start to diverge more, but unless Magento starts using the really new features like generated columns, window functions, JSON, common table expressions, etc. then the supported databases are basically "all of the above". To use these newer features we would either have to choose an officially supported flavor (my vote would be official MySQL) or write in conditionals of some sort (e.g. minor differences like VIRTUAL vs GENERATED).

Note, if you are using a really old version of MySQL I suggest you upgrade to at least 5.6 as there are so many improvements. The best way to upgrade is to setup replication and after the slave is caught up perform a failover.

**Comment by Quazz** [comment]
Have not had any problems upgrading (and switching from) mysql 5.5 all the way to MariaDB 10.2, so can confirm this works as expected. 

Newest PHP version that works is 7.0 (technically 7.1, but it throws warnings about mcrypt deprecation)

Redis is just the newest available version, afaik. 

So it's just PHP being annoying currently as far as I'm aware.

**Comment by seansan** [comment]
Php 7.2 is working over here /w the 1.9.4.x branch

Maybe update readme and close issue?

**Comment by tmotyl** [comment]
Somebody using Mariadb 10.3 or 10.4 with magelts?

**Comment by colinmollenhour** [comment]
I'm using MariaDb 10.2 with Galera with LTS, but I don't see any reason why 10.3 and 10.4 wouldn't also work just as well, we just haven't upgraded yet.

**Comment by tmotyl** [comment]
I'm asking because with Magento2 there were multiple performance issues with newer Mariadb, like https://github.com/magento/magento2/issues/25199

**Comment by colinmollenhour** [comment]
Ahh, good to know. Looks like MariaDb fixed the issue in 10.3.22, 10.4.12, 10.5.1.

**Comment by ufo56** [comment]
> Ahh, good to know. Looks like MariaDb fixed the issue in 10.3.22, 10.4.12, 10.5.1.

Weird index performance issues are still present with mariadb 10.4.12

**Comment by tmotyl** [comment]
@ufo56 do you have the same issues with 10.2 or 10.3? @colinmollenhour which version of 10.2 are you using with galera?

**Comment by ufo56** [comment]
> @ufo56 do you have the same issues with 10.2 or 10.3? @colinmollenhour which version of 10.2 are you using with galera?

10.3 was faster, no galera

**Comment by colinmollenhour** [comment]
Hmm, according to [MDEV-20900](https://jira.mariadb.org/browse/MDEV-20900) the fix was included in 10.4.12 as well. The bug is definitely not a bug in OpenMage, but we do appreciate reporting which versions are working or not working well.

**Comment by colinmollenhour** [comment]
> which version of 10.2 are you using with galera?

Have run many 10.1 versions and 10.2.26. The builds are here; https://hub.docker.com/r/colinmollenhour/mariadb-galera-swarm

**Comment by Flyingmana** [comment]
related: #1188

**Comment by luigifab** [comment]
We are using PHP 8.0 and MariaDB 10.5.9 with Master/Slave without problems since many weeks.
